File: ai-service/app/schemas/audit.py
from typing import Optional, Dict, Any
from datetime import datetime
from pydantic import BaseModel

class AuditLogResponse(BaseModel):
    id: str
    entity_type: str
    entity_id: str
    actor_id: str
    actor_name: Optional[str] = None
    actor_role: Optional[str] = None
    action: str
    details: Optional[Dict[str, Any]] = None
    timestamp: datetime

    class Config:
        from_attributes = True

File: backend/src/models/audit.rs
use chrono::NaiveDateTime;
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct AuditLog {
    pub id: String,
    pub entity_type: String,
    pub entity_id: String,
    pub actor_id: String,
    pub actor_name: Option<String>,
    pub actor_role: Option<String>,
    pub action: String,
    pub details: Option<serde_json::Value>,
    pub timestamp: Option<NaiveDateTime>,
}
